metis-dereference/metis-dereference-service/src/main/java/eu/europeana/metis/dereference/service/MongoDereferenceService.java

```diff
@@ -180,18 +180,18 @@ private void extractBroaderResources(EnrichmentBase resource, Set<String> destin
     }
 
     // If not in the cache, or no vocabulary (ID) is known, we need to resolve the resource.
-    final VocabularyCandidates candidates =
-            VocabularyCandidates.findVocabulariesForUrl(resourceId, vocabularyDao::getByUriSearch);
     if (entityString == null || vocabulary == null) {
+      final VocabularyCandidates candidates =
+          VocabularyCandidates.findVocabulariesForUrl(resourceId, vocabularyDao::getByUriSearch);
       final Pair<String, Vocabulary> transformedEntity =
           retrieveTransformedEntity(resourceId, candidates);
       if (transformedEntity != null) {
         entityString = transformedEntity.getLeft();
         vocabulary = transformedEntity.getRight();
-        final ProcessedEntity entityToCache = new ProcessedEntity();
+        ProcessedEntity entityToCache = (cachedEntity == null) ? new ProcessedEntity() : cachedEntity;
```

Comment:
Isn't this check done before? On line 176 if not mistaken. Why this if statement?

Reply:
Do you mean why the ternary operator is applied here?
Well yeah cachedEntity is checked on 176 but it can still be null here in which case we want a new ProcessedEntity object otherwise we want the non null cachedEntity.
That is done in that way to that the processedEntityDao.save(entityToCache); will know if it's an update(we are saving an object that we already have which has an identifier internally that is already known) or an insertion.